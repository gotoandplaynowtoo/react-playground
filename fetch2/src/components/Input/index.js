import styled from 'styled-components';

const Input = styled.input`
  border: 1px solid #aaa;
  padding: 4px;
`;

export default Input;

