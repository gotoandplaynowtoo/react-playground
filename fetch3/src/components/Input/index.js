import styled from 'styled-components';

const Input = styled.input`
  padding: 4px 6px;
  border: 1px solid #aaa;
`;

export default Input;