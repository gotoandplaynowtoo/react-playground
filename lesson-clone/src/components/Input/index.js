import styled from 'styled-components';

const Input = styled.input`
  border: 1px solid #aaa;
  display: inline-block;
  padding: 4px 6px;
`;

export default Input;