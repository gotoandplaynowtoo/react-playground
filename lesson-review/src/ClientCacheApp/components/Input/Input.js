import styled from 'styled-components';
import PropTypes from 'prop-types';

const Input = styled.input`
  padding: 4px 6px;
  border: 1px solid #eee;
`;

Input.propTypes = {
  children: PropTypes.node
};

export default Input;