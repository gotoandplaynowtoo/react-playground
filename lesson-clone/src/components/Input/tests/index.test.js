import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Input from 'components/Input';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Input/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('it renders correctly', () => {
  const tree = renderer.create(<Input/>).toJSON();
  expect(tree).toMatchSnapshot();
});