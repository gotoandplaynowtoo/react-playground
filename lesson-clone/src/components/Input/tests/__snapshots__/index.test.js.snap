// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<input
  className="sc-bdVaJa kSylSs"
/>
`;
